l1-contracts/src/core/messagebridge/Outbox.sol

```diff
@@ -60,21 +71,19 @@ contract Outbox is IOutbox {
    * @dev Emits `MessageConsumed` when consuming messages
    *
    * @param _message - The L2 to L1 message
-   * @param _l2BlockNumber - The block number specifying the block that contains the message we want to consume
-   * @param _leafIndex - The index inside the merkle tree where the message is located
-   * @param _path - The sibling path used to prove inclusion of the message, the _path length directly depends
-   * on the total amount of L2 to L1 messages in the block. i.e. the length of _path is equal to the depth of the
-   * L1 to L2 message tree.
+   * @param _epoch - The epoch that contains the message we want to consume
+   * @param _leafIndex - The index at the level in the wonky tree where the message is located
+   * @param _path - The sibling path used to prove inclusion of the message, the _path length depends
+   * on the location of the L2 to L1 message in the wonky tree.
    */
   function consume(
     DataStructures.L2ToL1Msg calldata _message,
-    uint256 _l2BlockNumber,
+    Epoch _epoch,
     uint256 _leafIndex,
     bytes32[] calldata _path
   ) external override(IOutbox) {
     require(_path.length < 256, Errors.Outbox__PathTooLong());
     require(_leafIndex < (1 << _path.length), Errors.Outbox__LeafIndexOutOfBounds(_leafIndex, _path.length));
-    require(_l2BlockNumber <= ROLLUP.getProvenBlockNumber(), Errors.Outbox__BlockNotProven(_l2BlockNumber));
```

Comment:
Pretty neat that we can avoid the call entirely here as it is just 0 before now 👍